src/bindings/js/node/src/async_infer_queue.cpp

```diff
@@ -0,0 +1,209 @@
+            m_requests[handle].set_callback([this, handle](std::exception_ptr exception_ptr) {
+                if (exception_ptr == nullptr) {
+                    auto ov_callback = [this](Napi::Env env, Napi::Function user_callback, int* handle) {
+                        Napi::Object js_ir = InferRequestWrap::wrap(env, m_requests[*handle]);
+                        const auto promise = m_user_ids[*handle].second;
+                        try {
+                            user_callback.Call({js_ir, m_user_ids[*handle].first.Value()});
+                            promise.Resolve(m_user_ids[*handle].first.Value());
+                            // returns before the promise's .then() is completed
+                        } catch (Napi::Error& e) {
+                            promise.Reject(Napi::Error::New(env, e.Message()).Value());
+                        }
+                        {
+                            // Start async inference on the next request or add idle handle to queue
+                            std::lock_guard<std::mutex> lock(m_mutex);
+                            if (m_awaiting_requests.size() > 0) {
+                                auto& request = m_awaiting_requests.front();
+                                start_async_impl(*handle,
+                                                 std::get<2>(request),
+                                                 std::get<0>(request).Value(),
+                                                 std::get<1>(request).Value());
+                                m_awaiting_requests.pop();
+                            } else {
+                                m_idle_handles.push(*handle);
+                            }
+                        }
+                        delete handle;
+                    };
+                    // The ov_callback will execute when the main event loop becomes idle
+                    m_tsfn.BlockingCall(new int(handle), ov_callback);
+                }
+                try {
+                    if (exception_ptr) {
+                        std::rethrow_exception(exception_ptr);
+                    }
+                } catch (const std::exception& e) {
+                    OPENVINO_THROW(e.what());
+                }
+            });
+        }
+    } catch (std::exception& e) {
+        reportError(info.Env(), e.what());
+    }
+}
+
+void AsyncInferQueue::start_async_impl(const int handle,
+                                       Napi::Promise::Deferred deferred,
+                                       Napi::Object infer_data,
+                                       Napi::Object user_data) {
+    m_user_ids[handle] = std::make_pair(Napi::Persistent(user_data), deferred);
+    m_user_inputs[handle] = Napi::Persistent(infer_data);  // keep reference to inputs so they are not garbage collected
+
+    // CVS-166764
+    const auto& keys = infer_data.GetPropertyNames();
+    for (uint32_t i = 0; i < keys.Length(); ++i) {
```

Comment:
```suggestion
    for (size_t i = 0; i < keys.Length(); ++i) {
```
?

Reply:
There are warnings on windows when using `size_t` as index for `Napi::Array`